core/src/com/unciv/models/ruleset/tech/Technology.kt

```diff
@@ -24,6 +24,20 @@ class Technology {
         val lineList = ArrayList<String>() // more readable than StringBuilder, with same performance for our use-case
         for (unique in uniques) lineList += unique.tr()
 
+        val mapOfImprovedImprovements = HashMap<String, ArrayList<String>>()
+        for (improvement in ruleset.tileImprovements.values) for ( unique in improvement.uniqueObjects
+                .filter { it.placeholderText in setOf("[] once [] is discovered", "[] on [] tiles once [] is discovered")
+                        && it.params.last() == name }) {
+            val key = if (unique.params.size == 2 ) unique.params[0] else "[${unique.params[0]}] on [${unique.params[1]}] tiles"
+            if (!mapOfImprovedImprovements.containsKey(key)) mapOfImprovedImprovements[key] = ArrayList()
+            mapOfImprovedImprovements[key]!!.add(improvement.name)
+        }
+        for ( improvements in mapOfImprovedImprovements) {
+            val impimpString = improvements.value.joinToString(", ") { it.tr() } +
+                    " {provide" + (if (improvements.value.size == 1) "s" else "") + "} " + improvements.key.tr()
```

Comment:
I really, really don't like this method of pluralization... there must be a way to have another sentence that doesn't require this.

Reply:
And yes I'm aware that this was probably copied from my code (impimpString rings a bell) but I was wrong, this is yukky

Reply:
There is this line:
![2020-09-01-22:54:50_1920x1080_scrot](https://user-images.githubusercontent.com/69697985/91900135-e3468f80-eca6-11ea-8931-c4c0322961a2.png)

So maybe
Stats [on tiles] from improvements: [list]
![2020-09-01-22:55:03_1920x1080_scrot](https://user-images.githubusercontent.com/69697985/91900151-e9d50700-eca6-11ea-951c-b077425ab1d3.png)

Reply:
Looks OK to me

Reply:
Conflicting files
android/assets/game.atlas
android/assets/game.png
android/assets/game2.png
did you merged branch with wrong atlas (from before river fix)?

Reply:
Resolved conflict